recipes/qt/5.x.x/conanfile.py

```diff
@@ -1064,17 +1064,13 @@ def _create_plugin(pluginname, libname, plugintype, requires):
             gui_reqs = []
             if self.options.with_dbus:
                 gui_reqs.append("DBus")
-            if self.options.with_freetype:
-                gui_reqs.append("freetype::freetype")
             if self.options.with_libpng:
                 gui_reqs.append("libpng::libpng")
             if self.options.get_safe("with_fontconfig", False):
                 gui_reqs.append("fontconfig::fontconfig")
             if self.settings.os in ["Linux", "FreeBSD"]:
                 if self.options.qtwayland or self.options.get_safe("with_x11", False):
                     gui_reqs.append("xkbcommon::xkbcommon")
-                if self.options.get_safe("with_x11", False):
-                    gui_reqs.append("xorg::xorg")
```

Comment:
xorg has to be referenced as a requirement of at least one component. Why do You think it's not needed here, wereas qt checks for these libraries during GUI module configuration https://github.com/qt/qtbase/blob/v5.15.10-lts-lgpl/src/gui/configure.json#L556 ?

Reply:
This dependencies is the first that I try to remove because I know that QtGUI is platform agnostic, and platform dependencies is in platforms plugins. To do this, I add test for QtGui module in test_package. If test can link and run, this mean that dependencies list defined in conanfile is exhaustive.

Now, I have a test to confirm that Xorg is not in QtGUI,

To be precise, xcb_qpa_lib is link against xcb, and platforms/xcb plugins (and platforms/xcb/gl_integrations) is link against xcb_qpa_lib
https://github.com/qt/qtbase/blob/v5.15.10-lts-lgpl/src/plugins/platforms/xcb/xcb_qpa_lib.pro#L94
https://github.com/qt/qtbase/blob/v5.15.10-lts-lgpl/src/plugins/platforms/xcb/xcb-plugin.pro#L3
https://github.com/qt/qtbase/blob/v5.15.10-lts-lgpl/src/plugins/platforms/xcb/gl_integrations/gl_integrations_plugin_base.pri#L1